hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDatanodeRegistration.java

```diff
@@ -79,8 +80,12 @@ public void checkConnect(String host, int port) {
   @Test
   public void testDNSLookups() throws Exception {
     MonitorDNS sm = new MonitorDNS();
-    System.setSecurityManager(sm);
-    
+    try {
+      System.setSecurityManager(sm);
+    } catch (UnsupportedOperationException e) {
+      assumeTrue("Test is skipped because SecurityManager cannot be set (JEP 411)", false);
```

Comment:
I would prefer to not lose test coverage, but I don't have a good idea for a different way to implement this.

Reply:
Neither did I.